### PR TITLE
Add support for multiline strings and unquoted keys

### PR DESCRIPTION
**Pull request by loreto**
Fork the hujson library from tailscale to add two changes to it:

1. Support multiline strings using backticks. This follows the notation for template literals in ECMAScript 6 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

2. Allows for unquoted keys when the key is a valid identifier. This is something that is also allowed by ECMAScript 6 and we're following almost the same rules (we're technically a little bit stricter because we do not allow `true`, `false` and `null` as they would complicate the implementation, and in practice I don't think we'll need those unquoted).

Added lots of new tests to make sure the library works with those changes.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,14 @@ module github.com/tailscale/hujson
...
+	github.com/google/go-cmp v0.5.8
+	github.com/k0kubun/pp v3.0.1+incompatible
+)
+
+require (
+	github.com/k0kubun/colorstring v0.0.0-20150214042306-9440f1994b88 // indirect
+	github.com/mattn/go-colorable v0.1.13 // indirect
+	github.com/mattn/go-isatty v0.0.16 // indirect
+	golang.org/x/sys v0.0.0-20220811171246-fbc7d0a398ab // indirect
+)
```
Comment (gcurtis) [inline go.mod]: `go mod tidy`?

File: standard.go [hunk 2]
```diff
@@ -83,3 +92,60 @@ func (b *Extra) standardize() {
...
+	}
+	if src.Kind() == 'a' {
+		return standarizedIdentifier(src)
+	}
+	return src
+}
+
+func standardizedMultiString(src Literal) Literal {
+	// Convert multiline strings into regular JSON strings:
+	result := make([]byte, 0, len(src)+10)
```
Comment (gcurtis) [inline standard.go:107]: Can `src` just be updated in-place to avoid a copy?

**Review by gcurtis — APPROVED** [review]
> I think this works for what we need, but there are a couple things we should probably note.
> 
> 1. Multi-line strings aren't parsed correctly when there's no space after a `:`. Here's a failing test:
> ```
> {
> 	in: "{a:`v`}",
> 	want: Value{
> 		StartOffset: 0,
> 		Value: &Object{
> 			Members: []ObjectMember{{
> 				Value{StartOffset: 1, Value: Literal("a"), EndOffset: 2},
> 				Value{StartOffset: 3, Value: Literal(`"v"`), EndOffset: 6},
> 			}},
> 		},
> 		EndOffset: 7,
> 	},
> 	wantStd: `{"a":"v"}`,
> }
> ```
> 

Wait I'm not sure we're on the same page. As written this test fails because ```Literal(`"v"`)``` indicates a double-quoted string in the JSON (but the input has a backtick string, not double quotes). If I change that to ```Literal("`v`")``` then it passes (which is what I expect)

File: standard.go (hunk 2 above)
Comment (loreto) [reply standard.go:107]: I don't think I can. When converting a multiline string to a "normal" string I need to replace newline characters by the characters `\` and `n` (to get the escape sequence `\n`). As a result the size of the buffer changes, and I don't know ahead of time how many newline characters I'll encounter.

File: go.mod (hunk 1 above)
Comment (loreto) [reply go.mod]: Done

**Pull request merged**